erigon-lib/common/cmp/cmp.go

```diff
@@ -17,11 +17,11 @@
 package cmp
 
 import (
-	"golang.org/x/exp/constraints"
+	"cmp"
```

Comment:
Should we still call our package `cmp` given the name clash?

Reply:
@antonis19 
what is your suggestion for naming?
I can think of `cmpext` for "cap extensions"

Reply:
or `cmpx`

Reply:
I'm fine with cmpx

Reply:
we have the same clash with std `math` vs `erigon-lib/common/math`, probably others too
seems like other people struggle with how to name packages that extend other packages - https://stackoverflow.com/questions/58476795/how-to-build-extensions-to-existing-packages-idiomatically
one suggestion there is to use `cmputil`

Reply:
I think I am actually fine with `cmp` also.  We could always rename when we do the import in case there is a class with the standard `cmp`. So I will approve and leave it up to you how you want to handle the name.

Reply:
yes, I think I'd prefer to leave it as `cmp` too for consistency
it seems that right now the way we handle clashes is by using aliases
so if std cmp clashes with erigon-lib/common/cmp the dev decides how to name the alias
we should probably have a standard for this but this is outside of the scope of this PR

Reply:
Yeah, I agree. Keeping consistent naming makes searching for these kind of packages simpler. And aliasing helps with the disambiguation.